console-api/proto/resources.proto

```diff
@@ -37,8 +37,10 @@ message Resource {
     common.MetaId metadata = 2;
     // The resources's concrete rust type.
     string concrete_type = 3;
+    // The location in code where the resource was created.
+    common.Location location = 4;
     // The kind of resource (e.g timer, mutex)
-    Kind kind = 4;
+    Kind kind = 5;
```

Comment:
nit: is there any reason we put the `location` field _before_ `Kind`? why not just add it afterwards, so we don't need to change the field number for the `Kind` field?
```protobuf
    // The kind of resource (e.g timer, mutex)
    Kind kind = 4;
    // The location in code where the resource was created.
    common.Location location = 5;
```

Changing the field number of an existing protobuf version will break protobuf backwards compatibility with earlier versions of the protobufs. Although we don't currently guarantee proto compatibility between two commits to this repo, we'll want to care about this once we start publishing releases of the console API, so it seems better to get in the practice of not re-numbering fields if we don't _need_ to...